repose-aggregator/tests/functional-test-framework/src/main/groovy/org/openrepose/framework/test/ReposeValveTest.groovy

```diff
@@ -127,6 +127,16 @@ abstract class ReposeValveTest extends Specification {
 
     }
 
+    def waitForHttpClientRequestCacheToClear(long timeoutMilliseconds = 500) {
```

Comment:
Love the method to document what this sleep is for.

Reply:
I will note that as of Repose 9, caching of shared requests can be disabled. To do so, set the `cache.ttl` attribute value to `0` in the `http-connection-pool.cfg.xml` file used for the test.

With that in mind, I would probably say that we should change the default `http-connection-pool.cfg.xml` file used in tests to disable shared request caching. Then for any tests that need that caching, we could set the `cache.ttl` attribute to a reasonable value. This is probably easiest to do by templating that value and giving it a default value of `0`. However, it would require identifying tests that need shared request caching. So that is probably outside the scope of this PR.

Reply:
Another thought I had was to have this method check the repose config file for the `cache.ttl` value and sleep only if the cache is enabled (e.g. cache.ttl > 0). But that would only save 500ms per call, and there can be multiple pools to slightly complicate things, so I didn't bother with that idea for now.